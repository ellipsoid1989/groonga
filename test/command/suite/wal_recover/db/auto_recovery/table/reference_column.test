# _database_close/_database_reopen can't use with HTTP because requests via HTTP are always
# processed by workers. Workers never use primary WAL role.
#@require-interface stdio

#@require-feature message_pack

#$GROONGA_ALLOW_DATABASE_REOPEN=yes
#$GRN_WAL_ROLE=primary

table_create Data TABLE_PAT_KEY ShortText

load --table Data
[
{"_key": "a"},
{"_key": "abcdefg"},
{"_key": "abcdefghijklmn"}
]

delete Data --key abcdefg

table_create Referer TABLE_HASH_KEY ShortText
column_create Referer data COLUMN_SCALAR Data

load --table Referer
[
{"_key": "123", "data": "a"}
]

dump

_database_close

#@copy-path #{db_path}.0000100 #{db_path}.0000100.wal
#@add-important-log-levels notice
_database_reopen
#@remove-important-log-levels notice

dump
