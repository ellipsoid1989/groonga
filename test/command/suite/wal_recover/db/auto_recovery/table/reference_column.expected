table_create Data TABLE_PAT_KEY ShortText
[[0,0.0,0.0],true]
load --table Data
[
{"_key": "a"},
{"_key": "abcdefg"},
{"_key": "abcdefghijklmn"}
]
[[0,0.0,0.0],3]
delete Data --key abcdefg
[[0,0.0,0.0],true]
table_create Referer TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create Referer data COLUMN_SCALAR Data
[[0,0.0,0.0],true]
load --table Referer
[
{"_key": "123", "data": "a"}
]
[[0,0.0,0.0],1]
dump
table_create Data TABLE_PAT_KEY ShortText

table_create Referer TABLE_HASH_KEY ShortText

column_create Referer data COLUMN_SCALAR Data

load --table Data
[
["_key"],
["a"],
["abcdefghijklmn"]
]

load --table Referer
[
["_key","data"],
["123","a"]
]
_database_close
[[0,0.0,0.0],true]
_database_reopen
[[0,0.0,0.0],true]
#|e| [reader][read-entry][pat][recover][Data] must be map: <positive-integer>(2): path:<db/db.0000100>
#|n| [db][wal][recover] rebuild broken table: <Data>(256)
#|n| DDL:259:table_create #recovering#Data
#|n| spec:259:update:#recovering#Data:49(table:pat_key):0
#|n| [db][wal][recover] rebuild broken column: <Referer.data>(258)
#|n| DDL:260:column_create Referer #recovering#data
#|n| spec:260:update:Referer.#recovering#data:64(column:fix_size):259(#recovering#Data)
#|n| DDL:258:obj_remove Referer.data
#|n| spec:258:remove:Referer.data:64(column:fix_size)
#|n| [db][wal][recover] succeeded to rebuild broken column: <Referer.data>(260)
#|n| DDL:256:obj_remove Data
#|n| spec:256:remove:Data:49(table:pat_key)
#|n| [db][wal][recover] succeeded to rebuild broken table: <Data>(259)
dump
table_create Data TABLE_PAT_KEY ShortText

table_create Referer TABLE_HASH_KEY ShortText

column_create Referer data COLUMN_SCALAR Data

load --table Data
[
["_key"],
["a"],
["abcdefghijklmn"]
]

load --table Referer
[
["_key","data"],
["123","a"]
]
